File: código/relative-avg.js
const {Mean} = require('./avg');
class RelativeMean extends Mean{
	get low_mean(){
		let mean = this.mean;
		return new this.constructor(this.filter(v => v < mean));
	}
	get high_mean(){
		let mean = this.mean;
		return new this.constructor(this.filter(v => v > mean));
	}
	get central_mean(){
		let low_mean = this.low_mean.mean, high_mean = this.high_mean.mean;
		return new this.constructor(this.filter(v => v > low_mean && v < high_mean));
	}
}
class RelativeGeometricMean extends RelativeMean{
	get mean(){
		return Math.pow(this.reduce((a, b) => a * b, 1), 1 / this.factor);
	}
}
class RelativeQuadraticMean extends RelativeMean{
	get mean(){
		return Math.sqrt(this.map(v => v ** 2).reduce((a, b) => a + b, 0) / this.factor);
	}
}
class RelativeHarmonicMean extends RelativeMean{
	get mean(){
		return this.factor / (this.map(v => v ** -1).reduce((a, b) => a + b, 0));
	}
}

module.exports = RelativeMean;
RelativeMean.RelGeoMean = RelativeGeometricMean;
RelativeMean.RelQuadMean = RelativeQuadraticMean;
RelativeMean.RelHarmMean = RelativeHarmonicMean;

File: código/relative-avg.py
from avg import Mean

class RelativeMean(Mean):
	def __init__(self, v*):
		Mean.__init__(v)
	def low_mean(self):
		mean = self.mean()
		return self.constructor(filter(lambda v: v < mean, self.values))
	def high_mean(self):
		mean = self.mean()
		return self.constructor(filter(lambda v: v > mean, self.values))
	def central_mean(self):
		low_mean = self.low_mean().mean(), high_mean = self.high_mean().mean()
		return self.constructor(filter(lambda v: v > low_mean and v < high_mean, self.values))
class RelativeGeometricMean(RelativeMean):
	def __init__(self):
		RelativeMean.__init__(self)
	def mean(self):
		prod = 1
		for v in self.values:
			prod *= v
		return pow(prod, 1 / self.factor())
class RelativeQuadraticMean(RelativeMean):
	def __init__(self):
		RelativeMean.__init__(self)
	def mean(self):
		return sqrt(sum(map(lambda v: v ** 2, self.values)) / self.factor())
class RelativeHarmonicMean(RelativeMean):
	def __init__(self):
		RelativeMean.__init__(self)
	def mean(self):
		return self.factor() / sum(map(lambda v: 1 / v, self.values))